src/scale.ts

```diff
@@ -10,44 +10,32 @@ import {Type, TYPE_INDEX} from './type';
 import {contains, Flag, keys} from './util';
 import {ScaleInterpolate, ScaleInterpolateParams} from './vega.schema';
 
-export namespace ScaleType {
+type ValueOf<T> = T[keyof T];
+export const ScaleType = {
   // Continuous - Quantitative
-  export const LINEAR: 'linear' = 'linear';
-  export const LOG: 'log' = 'log';
-  export const POW: 'pow' = 'pow';
-  export const SQRT: 'sqrt' = 'sqrt';
-  export const SYMLOG: 'symlog' = 'symlog';
+  LINEAR: 'linear',
+  LOG: 'log',
+  POW: 'pow',
+  SQRT: 'sqrt',
+  SYMLOG: 'symlog',
+
   // Continuous - Time
-  export const TIME: 'time' = 'time';
-  export const UTC: 'utc' = 'utc';
+  TIME: 'time',
+  UTC: 'utc',
 
   // Discretizing scales
-  export const QUANTILE: 'quantile' = 'quantile';
-  export const QUANTIZE: 'quantize' = 'quantize';
-  export const THRESHOLD: 'threshold' = 'threshold';
-  export const BIN_ORDINAL: 'bin-ordinal' = 'bin-ordinal';
+  QUANTILE: 'quantile',
+  QUANTIZE: 'quantize',
+  THRESHOLD: 'threshold',
+  BIN_ORDINAL: 'bin-ordinal',
 
   // Discrete scales
-  export const ORDINAL: 'ordinal' = 'ordinal';
-  export const POINT: 'point' = 'point';
-  export const BAND: 'band' = 'band';
-}
+  ORDINAL: 'ordinal',
+  POINT: 'point',
+  BAND: 'band'
+} as const;
 
-export type ScaleType =
-  | typeof ScaleType.LINEAR
-  | typeof ScaleType.LOG
-  | typeof ScaleType.POW
-  | typeof ScaleType.SQRT
-  | typeof ScaleType.SYMLOG
-  | typeof ScaleType.TIME
-  | typeof ScaleType.UTC
-  | typeof ScaleType.QUANTILE
-  | typeof ScaleType.QUANTIZE
-  | typeof ScaleType.THRESHOLD
-  | typeof ScaleType.BIN_ORDINAL
-  | typeof ScaleType.ORDINAL
-  | typeof ScaleType.POINT
-  | typeof ScaleType.BAND;
+export type ScaleType = ValueOf<typeof ScaleType>;
```

Comment:
This is just `keyof typeof` so let me see how I can support that in the schema generator.

Reply:
It's not. I was wrong.